zio-config/src/main/scala/zio/config/ConfigSource.scala

```diff
@@ -2,30 +2,115 @@ package zio.config
 
 import zio.config.ReadErrors.ReadError
 import zio.system.System
-import zio.{ IO, ZIO }
+import zio.{ IO, UIO, ZIO }
 
-trait ConfigSource[K, V] {
+trait ConfigSource[K, V] { self =>
   val configSourceService: ConfigSource.Service[K, V]
+
+  final def orElse(that: ConfigSource[K, V]): ConfigSource[K, V] =
+    new ConfigSource[K, V] {
+      val configSourceService: ConfigSource.Service[K, V] =
+        new ConfigSource.Service[K, V] {
+          val sourceDescription: String = {
+            val selfDesc = self.configSourceService.sourceDescription
+            val thatDesc = that.configSourceService.sourceDescription
+
+            s"${selfDesc}: with fallback source: ${thatDesc} (on all errors)"
+          }
+
+          def getConfigValue(path: Vector[K]): IO[ReadError[K, V], ConfigSource.Value[V]] =
+            self.configSourceService
+              .getConfigValue(path)
+              .orElse(
+                that.configSourceService
+                  .getConfigValue(path)
+              )
+        }
+    }
+
+  final def <>(that: ConfigSource[K, V]): ConfigSource[K, V] =
+    orElse(that)
+
+  final def catchMissingValue(that: => ConfigSource[K, V]): ConfigSource[K, V] =
```

Comment:
This is the same as `orElse`, except it only looks for `MissingValue`, not all `ReadError`. This may be a fringe use case and I think there is a more orthogonal way to handle this, anyway... may be worth dropping for now.

Reply:
Sure, I can drop this.